### PR TITLE
Move PsbtRpc to be able to be used by future versions of bitcoind

### PR DESCRIPTION
**Pull request by benthecarman**

### CODE REVIEW

File: bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/PsbtRpc.scala [hunk 1]
```diff
@@ -28,4 +28,12 @@ trait V18PsbtRpc {
     bitcoindCall[String]("utxoupdatepsbt", List(JsString(psbt)))
   }
 
+  def utxoUpdatePsbt(
```
Comment (nkohen) [inline bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/PsbtRpc.scala]: could you add a test for this new method

File: bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/DescriptorRpc.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  def deriveAddresses(
+      descriptor: String,
+      range: Option[Vector[Double]]): Future[DeriveAddressesResult] = {
+    bitcoindCall[DeriveAddressesResult](
+      "deriveaddresses",
+      List(JsString(descriptor), Json.toJson(range)))
+  }
+
+  def getDescriptorInfo(descriptor: String): Future[GetDescriptorInfoResult] = {
```
Comment (nkohen) [inline bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/DescriptorRpc.scala:29]: tests for these two

File: bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/DescriptorRpc.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  */
+trait DescriptorRpc {
+  self: Client =>
+
+  def deriveAddresses(
+      descriptor: String,
+      range: Option[Vector[Double]]): Future[DeriveAddressesResult] = {
+    bitcoindCall[DeriveAddressesResult](
+      "deriveaddresses",
+      List(JsString(descriptor), Json.toJson(range)))
```
Comment (nkohen) [inline bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/DescriptorRpc.scala]: In the case where `range == None` shouldn't this second param be omitted?

Comment (benthecarman) [reply bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/DescriptorRpc.scala]: Does `Json.toJson` do that automatically?

btw this is just copy paste from the original `v18DescriptorRpc.scala`

File: bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/DescriptorRpc.scala (hunk 2 above)
Comment (benthecarman) [reply bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/DescriptorRpc.scala:29]: They already exist in `BitcoindV18ClientTest`

File: bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/DescriptorRpc.scala (hunk 3 above)
Comment (nkohen) [reply bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/DescriptorRpc.scala]: `Json.toJson(None)` returns `JsNull`. Not sure if this is the intended functionality but I have a feeling that this param list should be a val defined on matching on `range`

**Review by nkohen — APPROVED** [review]
LGTM

**Pull request merged**